api/api.go

```diff
@@ -930,8 +931,16 @@ func parseWriteMeta(resp *http.Response, q *WriteMeta) error {
 
 // decodeBody is used to JSON decode a body
 func decodeBody(resp *http.Response, out interface{}) error {
```

Comment:
Should we always skip decoding body if `out == nil`?  I noticed that we wrap few `decodeBody` invocations with a nil check already:
https://github.com/hashicorp/nomad/blob/d890ddbfd9125f36bc7ff70cc7a584dd038e8432/api/api.go#L858-L862 .

It would simplify caller code a bit, and it doesn't seem like an error if caller explicitly passes nil and ignores response body (though body may need to be consumed... a potential bug in current code).